src/funcCoreTools/funcHostTask.ts

```diff
@@ -62,14 +66,33 @@ export function stopFuncTaskIfRunning(workspaceFolder: vscode.WorkspaceFolder):
     }
 }
 
-export function getFuncPortFromTask(funcTask: vscode.Task | undefined): string {
-    let port: string = defaultFuncPort;
+export async function getFuncPortFromTask(funcTask: vscode.Task | undefined, projectPathOrTaskScope: string | vscode.WorkspaceFolder | vscode.TaskScope): Promise<string> {
```

Comment:
Do you have to add `TaskScope` as a type here?  

It's kind of confusing since `vscode.TaskScope` is an enum of 1 or 2, but the scope property in task includes WorkspaceFolder.  I know that this is technically correct, but you're only using this parameter as a projectPath, so it seems like it'd make more sense to just ensure that it's a WorkspaceFolder [here](https://github.com/microsoft/vscode-azurefunctions/pull/2786/files?diff=split&w=1#diff-cef073404d3356451ab0401754e928cd82a926e6ec3fd4cab28f22d2d1adb8ceR37)

Reply:
> It's kind of confusing since vscode.TaskScope is an enum of 1 or 2, but the scope property in task includes WorkspaceFolder. 

I don't pick the types, VS Code does: https://github.com/microsoft/vscode/blob/69373b783347d93e598caef5be7a80099ba73983/src/vs/vscode.d.ts#L6641

> so it seems like it'd make more sense to just ensure that it's a WorkspaceFolder here

I don't really see the difference. Feels like 6 of one, half dozen of another

Reply:
> But I already approved it so if you feel like they're equivalent, then feel free to ignore.

I'd explain more of my reasoning if this was worth our time to debate. Seems like we both agree it's not worth our time, so yeah I'll go ahead and merge as-is